openedx/core/djangoapps/user_authn/views/registration_form.py

```diff
@@ -350,10 +350,11 @@ def __init__(self):
         field_order = configuration_helpers.get_value('REGISTRATION_FIELD_ORDER')
         if not field_order:
             field_order = settings.REGISTRATION_FIELD_ORDER or valid_fields
-
-        # Check that all of the valid_fields are in the field order and vice versa, if not set to the default order
+        # Check that all of the valid_fields are in the field order and vice versa,
+        # if not append missing fields at end of field order
         if set(valid_fields) != set(field_order):
-            field_order = valid_fields
+            difference = set(valid_fields).difference(set(field_order))
+            field_order.extend(difference)
```

Comment:
Why is this append necessary? If the confirm_email is enabled and the order is already defined, what are we going to achieve with this?

Reply:
There are some fields which are not in `REGISTRATION_FIELD_ORDER` as mentioned in description ie. "specialty", "profession" and if any of them are activated in `REGISTRATION_EXTRA_FIELDS` previous code was ignoring `REGISTRATION_FIELD_ORDER` and rendering fields as they were pushed in `valid_fields` list.